src/lib/sw_engine/tvgSwRaster.cpp

```diff
@@ -740,9 +748,11 @@ static bool _rasterScaledTranslucentRGBAImage(SwSurface* surface, const SwImage*
     if (image->scale < DOWN_SCALE_TOLERANCE) {
         for (auto y = region.min.y; y < region.max.y; ++y, dbuffer += surface->stride) {
             auto sy = static_cast<uint32_t>(y * itransform->e22 + itransform->e23);
+            if (sy >= image->h) continue;
             auto dst = dbuffer;
             for (auto x = region.min.x; x < region.max.x; ++x, ++dst) {
                 auto sx = static_cast<uint32_t>(x * itransform->e11 + itransform->e13);
```

Comment:
I think the problem occurs when the result of (x * itransform->e11 + itransform->e13) is negative.
(or memory problem ?)
Wouldn't the result of this expression be checked first(0 > result || w < result ?)  and then cast?

Reply:
we use "unsigned" so negative is not happening.

Reply:
As far as I know there is a problem casting float to "unsigned" int.
please refer to https://embeddeduse.com/2013/08/25/casting-a-negative-float-to-an-unsigned-int/

Reply:
@JSUYA  updated thanks.